addons/pos_stripe/static/src/js/payment_stripe.js

```diff
@@ -0,0 +1,234 @@
+        // We need the disconnect this reader if we want to use another one
+        if (
+            this.pos.connectedReader != this.payment_method.stripe_serial_number &&
+            this.terminal.getConnectionStatus() == 'connected'
+            ) {
+            this.terminal.disconnectReader().then(function(disconnectResult) {
+                if (disconnectResult.error) {
+                self._show_error(disconnectResult.error.message, disconnectResult.error.code);
+                line.set_payment_status('retry');
+                } else {
+                var connectReader = new Promise(function(resolve) {
+                    self.connect_reader(resolve);
+                });
+                connectReader.then(function(connected_reader) {
+                    if (connected_reader) {
+                        resolve(true);
+                    } else {resolve(false);}
+                });
+                }
+            });
+        } else if (this.terminal.getConnectionStatus() == 'not_connected') {
+                var connectReader = new Promise(function(resolve) {
+                    self.connect_reader(resolve);
+                });
+                connectReader.then(function(connected_reader) {
+                    if (connected_reader) {
+                        resolve(true);
+                    } else {resolve(false);}
+                });
+        } else {
+            resolve(true);
+        }
+    },
+
+    connect_reader(resolve) {
+        var self = this;
+        var line = this.pos.get_order().selected_paymentline;
+        var discoveredReaders = JSON.parse(this.pos.discoveredReaders);
+        for (const selectedReader of discoveredReaders) {
+            if (selectedReader.serial_number == this.payment_method.stripe_serial_number) {
+                this.terminal.connectReader(selectedReader, {fail_if_in_use: true}).then(function(connectResult) {
+                    if (connectResult.error) {
+                    self._show_error(connectResult.error.message, connectResult.error.code);
+                    line.set_payment_status('retry');
+                    line.hideDeleteButton = false;
+                    resolve(false);
+                    } else {
+                    console.log('Connected to reader: ', connectResult.reader.label);
+                    self.pos.connectedReader = self.payment_method.stripe_serial_number;
+                    resolve(true);
+                    }
+                });
+            }
+        }
+    },
+
+    collect_payment(resolve, reject, currency, amount) {
+        var self = this;
+        var line = this.pos.get_order().selected_paymentline;
+        this.fetch_payment_intent_client_secret(currency, amount).then(function(client_secret) {
+            line.set_payment_status('waitingCard');
+        self.terminal.collectPaymentMethod(client_secret, {config_override: {skip_tipping: false}}).then(function(result) {
```

Comment:
From what I understand, stripe terminal is not like our adyen terminal that is asynchronous where getting the result of the payment request requires the adyen server to contact our server.

Here, we just have to call `collectPaymentMethod`, wait until customer confirms the payment, then call `processPayment`, followed by the final result -- all in single async function call. This is good. It means there is no need for us to poll for result, stripe is doing it for us from the terminal.

I think this code can be simpler if you declare the methods to be `async`, that way, you don't need to pass around the resolve method from `send_payment_request`.